src/test/java/StringManipulationTest.java

```diff
@@ -53,59 +62,88 @@ public void testRemoveNthCharacter2() {
     }
 
+    //This test checks n = 1 case removes all spacing.
     @Test
     public void testRemoveNthCharacter4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
+        assertEquals("", manipulatedstring.removeNthCharacter(1, false));
     }
 
+    //This test check that an Indexoutofbounds exception is thrown appropriately.
     @Test
     public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
+        assertThrows(IndexOutOfBoundsException.class, () -> manipulatedstring.removeNthCharacter(50, false));
     }
 
+    //This test check that an IllegalArgumentException exception is thrown appropriately.
     @Test
     public void testRemoveNthCharacter6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
+        assertThrows(IllegalArgumentException.class, () -> manipulatedstring.removeNthCharacter(0, false));
     }
 
+    //This test check that an IllegalArgumentException exception is thrown appropriately.
     @Test
     public void testRemoveNthCharacter7() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
+        assertThrows(IllegalArgumentException.class, () -> manipulatedstring.removeNthCharacter(-1, false));
     }
 
     @Test
-    public void testGeSubStrings1() {
+    public void testGetSubStrings1() {
         manipulatedstring.setString("This is my string");
-        String [] sStings = manipulatedstring.getSubStrings(3, 4);
+        String [] sStrings = manipulatedstring.getSubStrings(3, 4);
 
-        assertEquals(sStings[0], "my");
-        assertEquals(sStings[1], "string");
+        assertEquals(sStrings[0], "my");
+        assertEquals(sStrings[1], "string");
     }
 
+    //This test handles larger given strings
     @Test
-    public void testGeSubStrings2() {
-        fail("Not yet implemented");
+    public void testGetSubStrings2() {
+        manipulatedstring.setString("The Greatest String In The Universe");
+        String [] sStrings = manipulatedstring.getSubStrings(3, 6);
+
+        assertEquals(sStrings[0], "String");
+        assertEquals(sStrings[1], "In");
+        assertEquals(sStrings[2], "The");
+        assertEquals(sStrings[3], "Universe");
     }
+
+    //This test handles IllegalArgumentException for improper start and end values
     @Test
-    public void testGeSubStrings3() {
-        fail("Not yet implemented");
+    public void testGetSubStrings3() {
+        manipulatedstring.setString("This is my string");
+        assertThrows(IllegalArgumentException.class, () -> manipulatedstring.getSubStrings(3, 2));
     }
+
+    //This test handles IllegalArgumentException for improper start and end values
     @Test
```

Comment:
This test case is redundant. you're Testing the situation where the startWord is greater than the endWord, but there is also the possibility the start word and end words are the same, OR that the parameter inputs a negative number on accident. These are other cases you can test.